api/envoy/api/v2/auth/cert.proto

```diff
@@ -391,6 +391,9 @@ message DownstreamTlsContext {
     // Config for fetching TLS session ticket keys via SDS API.
     SdsSecretConfig session_ticket_keys_sds_secret_config = 5;
   }
+
+  // If specified, session_timeout is used as a hint for tls tickets lifetime
+  google.protobuf.UInt32Value session_timeout = 6;
```

Comment:
Should this be `google.protobuf.Duration`?

Reply:
looking on https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration. Duration returns seconds in int64, but boringssl expects session_timeout to be in uint32. obviously we could do cast (+border values checking)  in code, if we would decide to go w/ duration, if you have strong feelings against uint32.

Reply:
I think we should use duration types for the API surface. We can document any limitations (including via protoc-gen-validate constraints).